### PR TITLE
update(docs): add overview and versioning sections to falco release.md

### PR DESCRIPTION
**Pull request by incertum**
Co-authored-by: Leonardo Grasso <me@leonardograsso.com>
Signed-off-by: Melissa Kilby <melissa.kilby.oss@gmail.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

/kind documentation

> /kind failing-test

> /kind feature

> /kind release

> If contributing rules or changes to rules, please make sure to also uncomment one of the following line:

> /kind rule-update

> /kind rule-create

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area engine

> /area rules

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:

Add more technical clarity to the release doc, see https://github.com/falcosecurity/falco/issues/2195

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

@leogr - started with adding an overview section and as discussed more information around versioning of Falco components. Perhaps we should adjust the old sections to the new style you started for libs https://github.com/falcosecurity/libs/pull/580? Would you want to take care of the next commit and I can then help polish things afterwards? Please feel free to overwrite and improve anything I wrote so far :)

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,18 +1,69 @@
...
+## Overview
+
+This document provides the process to create a new Falco release. In addition, it provides information about the versioning of the Falco components. At a high level each Falco release consists of the following main components:
+
+- Falco binary (userspace)
+- Falco kernel driver object files (kernel space)
+    - Option 1: Kernel module (`.ko` files)
+    - Option 2: eBPF (`.o` files)
+- Falco config and rules `.yaml` files (userspace)
+- Falco plugins (userspace - optional)
```
Comment (leogr) [inline RELEASE.md:13]: Actually, some plugins (`k8saudit` and `cloudtrail`) are shipped together with Falco. 
That indirectly implies:
- plugins have their own versioning and [release strategy](https://github.com/falcosecurity/plugins/blob/master/release.md)
- If necessary, those plugins must be released before the Falco release, then their versions bumped in the related [cmake module](https://github.com/falcosecurity/falco/blob/master/cmake/modules/plugins.cmake) (it's similar to what we do with `libs` and `drivers`)
- Versioning: the plugin API version must be compatible

File: RELEASE.md [hunk 2]
```diff
@@ -1,18 +1,69 @@
...
+Alternatively Falco binaries or plugins can also be downloaded from the Falco Artifacts repo.
+
+- [Falco Artifacts Repo Packages Root](https://download.falco.org/?prefix=packages/)
+- [Falco Artifacts Repo Plugins Root](https://download.falco.org/?prefix=plugins/)
+
+
+### Falco Drivers Artifacts Repo - Quick Links
+
+
+The Falco project publishes all drivers for each release for all popular kernel versions / distros and `x86_64` and `aarch64` architectures to the Falco project managed Artifacts repo. The Artifacts repo follows standard directory level conventions. The respective driver object file is prefixed by distro and named / versioned by kernel release - `$(uname -r)`. More details around driver versioning and driver compatibility are provided in the [Falco Components Versioning](#falco-components-versioning) section.
```
Comment (leogr) [inline RELEASE.md:44]: Perhaps it is worth specifying that:
 - pre-builts need to be published before the Falco release
 - pre-builts are provided on a [best-effort basis](https://github.com/falcosecurity/falco/blob/master/proposals/20200818-artifacts-storage.md#notice)

File: RELEASE.md [hunk 3]
```diff
@@ -121,3 +172,19 @@ Announce the new release to the world!
...
 - IFF the on going release introduces a **new minor version**, [archive a snapshot of the Falco website](https://github.com/falcosecurity/falco-website/blob/master/release.md#documentation-versioning)
+
+
+## Falco Components Versioning
+
+This section provides more details around the versioning of all components that make up core Falco. It can also be a useful guide for the uninitiated to be more informed about Falco's source. Because the `libs` repo contains >90% of Falco's core features and is the home of each of the kernel drivers and engines, the [libs release doc](https://github.com/falcosecurity/libs/blob/master/release.md) is an excellent additional resource. `SHA256` checksums are provided throughout Falco's source code to empower the end user to perform integrity checks. All Falco releases also contain the sources as part of the packages.
+
+
+### Falco repo (this repo)
+- Falco version is a git tag (`x.y.z`), see [Procedures](#procedures) section.
```
Comment (leogr) [inline RELEASE.md:183]: Not sure if we should mention that it's a sem-ver-like schema, but not fully compatible with sem-ver

**Review by leogr — COMMENTED** [review]
Nice :star_struck: 

Just left some minor comments after a first quick look :smiley:

**Comment by leogr** [comment]
> @leogr - started with adding an overview section and as discussed more information around versioning of Falco components. Perhaps we should adjust the old sections to the new style you started for libs [falcosecurity/libs#580](https://github.com/falcosecurity/libs/pull/580)? Would you want to take care of the next commit and I can then help polish things afterwards? Please feel free to overwrite and improve anything I wrote so far :)

:+1: 

I plan to do that once we have released `libs` (so we at least can say we have tried it :smile_cat: )

**Comment by incertum** [comment]
@leogr Added additional details into the doc as suggested. Ok to ask for more polishing commit. After that we can merge and you can edit the rest in a follow up PR.

**Comment by leogr** [comment]
I guess this PR will remain in WIP until the release, so
/milestone 0.34.0

**Review by leogr — APPROVED** [review]

**Review by FedeDP — APPROVED** [review]
Love this, really great effort!
/approve

**Pull request merged**